src/emulation/sync.ts

```diff
@@ -452,28 +452,75 @@ export function readdirSync(
 	} catch (e) {
 		throw fixError(e as Error, { [resolved]: path });
 	}
+
+	// Check for mounted points
 	for (const mount of mounts.keys()) {
 		if (!mount.startsWith(path)) {
 			continue;
 		}
 		const entry = mount.slice(path.length);
-		if (entry.includes('/') || entry.length == 0) {
-			// ignore FSs mounted in subdirectories and any FS mounted to `path`.
-			continue;
+		if (entry.includes('/') || entry.length === 0) {
```

Comment:
This change isn't really needed. It's totally fine, though I'd prefer if changes like this were minimized in the future to keep the blame (as in `git blame`) clean.